docs/source/data/subscriptions.mdx

```diff
@@ -366,159 +393,154 @@ subscription IncrementingNumber {
+    return { currentUser };
+  }
+  // Otherwise let our resolvers know we don't have a current user
+  return { currentUser: null };
+};
 
-</td>
-</tr>
+useServer(
+  {
+    schema,
+    // Adding a context property lets you add data to your GraphQL operation context
+     context: (ctx, msg, args) => {
+       // Returning an object will add that information to our
+       // GraphQL context, which all of our resolvers have access to.
+      return getDynamicContext(ctx, msg, args);
+    },
+  },
+  wsServer,
+);
+```
+Putting it all together, if the `useServer.context` function returns an _object_ that object is passed to the GraphQL operation `context`, which is available to your resolvers.
 
-<tr>
-<td>
+Note that the `context` option is called once per `subscription` request, not once per event emission. Meaning that in the above example, every time a client sends a `subscription` operation we would check their authentication token.
 
-###### `context`
+### `onConnect` and `onDisconnect`
 
-`ConnectionContext`
-</td>
-<td>
+Expanding on the above, you can add additional options for the subscription server to execute whenever a client connects (`onConnect`) or disconnects (`onDisconnect`).
 
-Context object for the WebSocket connection. This is _not_ the `context` object for the associated subscription operation.
+Defining an `onConnect` function lets you reject a particular incoming connection by returning false or by throwing an exception. This can be helpful if you'd like to check authentication when a client first connects to your subscription server.
 
-[See fields](https://github.com/apollographql/subscriptions-transport-ws/blob/master/src/server.ts#L35-L43)
+You provide these functions as options to the first argument of `useServer`, like so:
 
-</td>
-</tr>
+```ts{5,12}
+useServer(
+  {
+    schema,
+    // As before, ctx is the graphql-ws Context where connectionParams live.
+    onConnect: async (ctx) => {
+      // Check authentication every time a client connects.
+      if (tokenIsNotValid(ctx.connectionParams)) {
+        // You can return false to close the connection  or throw an explicit error
+        throw new Error('Auth token missing!');
+      }
+    },
+    onDisconnect(ctx, code, reason) {
+      console.log('Disconnected!');
+    },
+  },
+  wsServer,
+);
+```
+For more information and examples of using `onConnect` and `onDisconnect`, see the [`graphql-ws` documentation](https://github.com/enisdenjo/graphql-ws#ws-auth-handling).
 
-</tbody>
-</table>
+### Example: Authentication with the Server and Client
 
-### Example: Authentication with `onConnect`
+>If you plan to use subscriptions with Apollo Client, ensure **[both your client and server subscription protocols are consistent](https://www.apollographql.com/docs/react/data/subscriptions/#choice-of-subscription-protocol)** for the subscription library you are using (this example uses the `graphql-ws` library).
```

Comment:
This is an important-enough consideration that it might be a good idea to have a dedicated h2 section called like "Configuring clients" that first states at a higher level that you want to make sure all of your clients use the `graphql-ws` protocol. And then has a bulleted list item for each Apollo client library that either links to relevant configuration _or_ says that no configuration is required

Reply:
Ah ok so this is covered in the bottom section! Maybe just have this notice link down there since it's a pretty good summary of considerations?

Reply:
Such a great idea, thank you!